src/Symfony/Component/Form/Form.php

```diff
@@ -505,10 +505,27 @@ public function submit($submittedData, bool $clearMissing = true)
         // they are collectable during submission only
         $this->errors = [];
 
+        $dispatcher = $this->config->getEventDispatcher();
+
         // Obviously, a disabled form should not change its data upon submission.
-        if ($this->isDisabled()) {
+        if ($this->isDisabled() && $this->isRoot()) {
```

Comment:
Sorry for late feedback, but, if I'm not missing something, I think this change will break the current behavior.

Assuming you have a child field disabled, getting its value this way `$form->get('field')->getData()` will return the submitted data rather than its initial value as expected.

It is likely that the new condition '$this->isRoot()` should group the new code in place, thus keeping the previous behavior of ignoring disabled child forms.